resources/examples/manager/BasicAssetLibrary/plugin/BasicAssetLibrary/BasicAssetLibraryInterface.py

```diff
@@ -120,11 +124,22 @@ def resolve(self, entityRefs, traitSet, context, hostSession):
                 for trait in traitSet:
                     trait_data = entity.traits.get(trait)
                     if trait_data is not None:
-                        result.addTrait(trait)
-                        for property_, value in trait_data.items():
-                            result.setTraitProperty(trait, property_, value)
+                        self.__add_trait_to_traits_data(result, trait, trait_data)
             except Exception as exc:  # pylint: disable=broad-except
                 exc_name = exc.__class__.__name__
                 result = EntityResolutionError(f"{exc_name}: {exc}", ref)
             results.append(result)
         return results
+
+    @classmethod
+    def __dict_to_traits_data(cls, traits_dict: dict):
+        traits_data = TraitsData()
+        for trait_id, trait_properties in traits_dict.items():
+            cls.__add_trait_to_traits_data(traits_data, trait_id, trait_properties)
+        return traits_data
+
+    @staticmethod
+    def __add_trait_to_traits_data(traits_data: TraitsData, trait_id: str, trait_properties: dict):
```

Comment:
[question] Not so critical in the context of BAL, but I _think_ we have a general convention in the code base that for a method `do_a_to_b` the args would be `do_a_to_b(a, b)` rather than `do_a_to_b(b, a)`. So the args match the sentence. Any thoughts on this, and whether we should codify it in our Coding standards? Consistency here in `OpenAssetIO` at least is quite important for familiarity.

Reply:
Good point, updated in 910ec973